From 7aa0ba29e501156c8e8577dd39186634ffc0b4d0 Mon Sep 17 00:00:00 2001
From: Libin Yang <libin.yang@intel.com>
Date: Fri, 24 May 2019 11:03:07 +0800
Subject: [PATCH] FROMGIT: ASoC: hdac_hdmi: jack hotplug in S3

BugLink: https://bugs.launchpad.net/bugs/1826181

Signed-off-by: Libin Yang <libin.yang@intel.com>
(cherry picked from commit d596fe8cc6f46b4fd315818ca25db0e785981834
git://github.com/thesofproject/linux.git)
Signed-off-by: Hui Wang <hui.wang@canonical.com>
---
 sound/soc/codecs/hdac_hdmi.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/sound/soc/codecs/hdac_hdmi.c b/sound/soc/codecs/hdac_hdmi.c
index b4334bcb891e..0ae9073d56ee 100644
--- a/sound/soc/codecs/hdac_hdmi.c
+++ b/sound/soc/codecs/hdac_hdmi.c
@@ -189,7 +189,7 @@ static void hdac_hdmi_jack_report(struct hdac_hdmi_pcm *pcm,
 			snd_soc_jack_report(pcm->jack, SND_JACK_AVOUT,
 						SND_JACK_AVOUT);
 		}
-		pcm->jack_event++;
+		pcm->jack_event = 1;
 	} else {
 		/*
 		 * Report Jack disconnect event when a device is disconnected
@@ -198,8 +198,7 @@ static void hdac_hdmi_jack_report(struct hdac_hdmi_pcm *pcm,
 		 */
 		if (pcm->jack_event == 1)
 			snd_soc_jack_report(pcm->jack, 0, SND_JACK_AVOUT);
-		if (pcm->jack_event > 0)
-			pcm->jack_event--;
+		pcm->jack_event = 0;
 	}
 
 	snd_soc_dapm_sync(port->dapm);
-- 
2.17.1

